ambari-server/src/main/java/org/apache/ambari/server/orm/entities/ServiceDesiredStateEntity.java

```diff
@@ -25,23 +25,32 @@
 import javax.persistence.Enumerated;
 import javax.persistence.Id;
 import javax.persistence.JoinColumn;
-import javax.persistence.JoinColumns;
+import javax.persistence.NamedQueries;
+import javax.persistence.NamedQuery;
 import javax.persistence.OneToOne;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
 
 import org.apache.ambari.server.state.MaintenanceState;
 import org.apache.ambari.server.state.State;
 
-@javax.persistence.IdClass(ServiceDesiredStateEntityPK.class)
-@javax.persistence.Table(name = "servicedesiredstate")
+@NamedQueries({
+        @NamedQuery(name = "ServiceDesiredStateByServiceId", query =
+                "SELECT serviceDesiredState " +
+                        "FROM ServiceDesiredStateEntity serviceDesiredState " +
+                        "WHERE serviceDesiredState.serviceId=:serviceId ")
+})
+@Table(
+        name = "servicedesiredstate",
+        uniqueConstraints = @UniqueConstraint(name = "UQ_servicedesiredstate",
+                                              columnNames = {"service_id"}))
 @Entity
```

Comment:
Spacing here looks off - 4 spaces instead of 2?

Reply:
Done.